picard/script.py

```diff
@@ -59,7 +59,14 @@ class ScriptError(Exception):
 
 
 class ScriptParseError(ScriptError):
-    pass
+    def __init__(self, message, stackitem):
+        super().__init__(
+            "{0} at position {1}, line {2}".format(
+                message,
+                stackitem.line,
+                stackitem.column
```

Comment:
Yeah, that's a habit I need to develop.  I've gone through my code and tried to change all occurrences to use named arguments.

Reply:
> So I would unify to:
`"{line}:{column}: {message}"` and `"{line}:{column}:{name}: {message}"`

I can do that, but that would mean that the error message shown when entering a script would change from (for example) "**Unexpected character at line 10, column 27**" to "**10:27: Unexpected character**".

I'm okay with that because I understand what the "10:27" means, but I wonder if other users will find this confusing.

For now, I've set up all exceptions to use the `"{line}:{column}: {message}"` and `"{line}:{column}:{name}: {message}"` format, including those displayed to the user during script development / input.

Also moved the message text for the `ScriptEndOfFile` and `ScriptUnknownFunction` exceptions to the exceptions themselves so that is only appears in one place, since there is a standard message for these two exceptions.

Reply:
> I'm okay with that because I understand what the "10:27" means, but I wonder if other users will find this confusing.

The idea behind this is to make messages consistent, and ease translation, the final format of the message shouldn't prevent us to improve this.
Of course "Unexpected character at line 10, column 27" is clearer than "10:27: Unexpected character", and if you prefer keep it, anything til formats of all message are consistent.

Reply:
Left as  `"{line}:{column}: {message}"` and `"{line}:{column}:{name}: {message}"`.  We can always change it later if enough people notice and complain.